addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -83,6 +83,7 @@ def _validate_leave_request(self):
                     'date': fields.Date.to_string(day_date),
                     'holiday_id': holiday.id,
                     'employee_id': holiday.employee_id.id,
+                    "company_id": holiday_project.company_id.id,
```

Comment:
In master we fixed this with "holiday_task.company_id.id or holiday_project.company_id.id". Do you think it can solve your problem too ?

Reply:
@jem-odoo Of course. Backport it to 12.3 then, thanks.